docs/IR.md

```diff
@@ -324,7 +331,11 @@ For example, a NxM matrix would have the shape list [N,M].
 
 The name of each dimension variable MUST adhere to C identifier syntax.
 
-Dimension variables are scoped to the declaration (graph signature, node signature, or single operator declaration) that they appear in. Thus, any given name denotes the same value within a declaration, allowing a declaration to describe how the shapes of inputs and outputs are related. For example, a graph that performs matrix cross-product may be defined as taking two inputs of shape [K,M] and [M,N], and produce an output of shape [K,N].
+Dimension variables are scoped to the declaration that they appear in. For graph inputs and outputs, the graph itself is the declaration. Dimension variables appearing in a graph's 'value_info' record are scoped to the value. Consequently, any name that is repeated denotes the same value within a declaration, allowing a declaration to describe how the shapes of inputs and outputs are related.
```

Comment:
Does it mean that the dimensions defined by a 'value_info"s are mutually independent even if some of them are identical strings?

Reply:
What I'm trying to say here, perhaps not precisely enough, is that for graph inputs and outputs, dimension variables are scoped to the collection of inputs and outputs. This is so that we can associate dimension information between inputs and outputs. For value_info records that are not inputs or outputs, I believe we don't force association across values, but maybe that's a misunderstanding on my part.

Reply:
I believe you're right but this might need help from all praters to confirm. To make this concept clearer, you might want to change **"Dimension variables appearing in a graph's 'value_info' record are scoped to the value"** to **"Dimension variables appearing in a graph's 'value_info' record are scoped to the value so they don't force any association across values which are not graph's inputs or outputs"**